gtsfm/frontend/verifier/homography.py

```diff
@@ -0,0 +1,71 @@
+from typing import Tuple
+
+import cv2
+import numpy as np
+
+from gtsfm.common.keypoints import Keypoints
+
+# from gtsfm.frontend.verifier.verifier_base import TwoViewEstimationReport
+
+import gtsfm.utils.logger as logger_utils
+
+logger = logger_utils.get_logger()
+
+"""
+Verification that determines whether the scene is planar or the camera motion is a pure rotation.
+
+COLMAP also checks degeneracy of structure here:
+    https://github.com/colmap/colmap/blob/dev/src/estimators/two_view_geometry.cc#L277
+"""
+
+MIN_PTS_HOMOGRAPHY = 4
+DEFAULT_RANSAC_PROB = 0.999
+
+
+class HomographyEstimator:
+    def __init__(self, estimation_threshold_px: float) -> None:
+        """
+        """
+        self._px_threshold = estimation_threshold_px
+
+    def estimate(
+        self, keypoints_i1: Keypoints, keypoints_i2: Keypoints, match_indices: np.ndarray
+    ) -> Tuple[np.ndarray, float, int]:
+        """Estimate to what extent the correspondences agree with an estimated homography.
```

Comment:
As we are returning the H matrix too, we are essentially doing the full homography estimation, with the percentage of agreement being one of the things computed. I think the header line should be 'Fit a homography to the correspondences`. Or whatever we use for the corresponding function in the verifier.

Reply:
good point, updated